File: May Long Two  2022/XOR_PAL.py
# cook your dish here

for _ in range(int(input())):
    N=int(input())
    A = input()[:N]
    # print(A)
    B=A[::-1]
    # print(B)
    count=0
    for i in range(N//2):
        if A[i]!=B[i]:
            count+=1
            
    # print(count)
    ans=0
    if(count%2==0):
        
        ans=count//2
        
    else:
        ans=count//2+1
        
    print(ans)
    

File: XOR_PAL.cpp
#include <iostream>
using namespace std;

int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--){
	    int n;
	    cin>>n;
	    string s;
	    cin>>s;
	    int count=0;
	    for(int i=0;i<n/2;i++)
	        count+= (s[i] != s[n-i-1]);
	        
	        cout << (count < 2 ? count : count / 2 + count%2) <<endl;
	    
	}
	return 0;
}
